### PR TITLE
Feat(Paywalls) Support min/max sizing

### PR DESCRIPTION
**Pull request by JZDesign**
### Checklist
- [x] If applicable, unit tests
- [ ] If applicable, create follow-up issues for `purchases-ios` and hybrids

### Motivation
Paywalls v2 components can declare `min`/`max` bounds on `Fit` and `Fill` sizes (Linear: PW-1359). The Android renderer ignored them: `Modifier.size` only mapped to `wrapContent`/`fillMax*`/`width()`, stacks distributed `Fill` space purely by weight, and media/WebView intrinsic sizing had no way to clamp. This brings Android to parity with the dashboard preview and iOS.

### Description

**Sizing primitives**
- `Modifier.size` now measures each axis with explicit limits. With no `min`/`max` set it is behaviourally identical to the previous `wrapContent`/`fillMax*`/`width()` modifiers, which is why existing snapshots are unchanged.
- `Size.addMargin` consolidates the three copies of "grow the size by the margin" (stack, image, video, icon) and now also applies to `Fill`/`Fit` limits, not just `Fixed`.
- `Size.adjustForMedia` consolidates image/video intrinsic sizing. When both axes are `Fit`, the media is scaled uniformly to satisfy both axes' limits, so aspect ratio is preserved.
- WebView `Fit`/unbounded-`Fill` resolution clamps the measured content size.

**Stacks**
- Children carry their *resolved* size (overrides + margin applied) to the parent via `ComponentSizeParentDataModifier`, so `ConstrainedFillLayout` allocates space from the size the child will actually enforce, matched by parent data rather than by index (robust to `visible = false` children).
- Any stack with a `Fill` child on its main axis now goes through `ConstrainedFillLayout`. Its measuring and rounding mirror `Row`/`Column` exactly, so unconstrained paywalls render identically (verified below); constrained `Fill`s get their `min`/`max` honoured with overflow handled sequentially instead of overlapping.
- A `Fit` stack with a positive `min` and a `SPACE_*` distribution distributes the extra space instead of collapsing.
- The root-stack "force `Fill`" hack now preserves any `min`/`max` that was declared.

**Resolved semantics for ambiguous cases**
- `min` and `max` conflicting (e.g. `Fill(min = 100)` in an 80dp parent): `min` wins and the component overflows, matching the existing overflow behaviour.
- Media with one axis `Fixed` and the other `Fit(max)`: the `max` wins over the intrinsic aspect ratio; content scale decides crop vs letterbox.
- `Fill` with `max` in a stack no longer consumes a full weighted share; the remainder goes to unconstrained siblings or the distribution.

### Testing
- New unit tests: `SizeModifierTest`, `SizeTest`, `MediaSizeTest`, `WebViewEffectiveSizeTest`, `StackSizeConstraintTest`, `StackConstrainedFillDistributionTest` (including override-only constraints), `StackFillUnboundedCollapseTest`.
- Emerge: 21 new preview snapshots (`StackMinMaxPreviews`, `SizeRtlPreviews`), 0 existing snapshots modified.
- Local Paparazzi: all 131 real-paywall template snapshots (`PaywallComponentsTemplatePreviewRecorder`) are pixel-identical to `main`.

---

> [!NOTE]
> **Medium Risk**
> Touches core paywall layout and measurement for all component types; regressions could affect sizing, scrolling, and RTL, though coverage is extensive.
> 
> **Overview**
> Adds **min/max bounds** to paywall component sizing (`Fit`, `Fill`, and margins) across stacks, media, and WebViews.
> 
> The custom **`Modifier.size`** path now measures with axis limits (including RTL alignment) instead of only `wrapContent` / `fillMax*`. **`adjustForMedia`** centralizes image/video intrinsic sizing when both axes are `Fit`, scaling together to honor min/max while keeping aspect ratio. Margins bump **Fill/Fit** limits via shared **`addMargin`**, and components expose resolved sizes to parents through **`ComponentSizeParentDataModifier`**.
> 
> **Stack layout** switches to **`ConstrainedFillLayout`** when children use bounded `Fill`, or when the stack has a positive **`Fit` minimum** with space-based distributions; allocation respects min/max, margins, overflow, and capped `Fill` no longer always consuming a full weighted slot. The root-stack “force Fill” hack now **preserves** existing min/max when promoting `Fit` to `Fill`.
> 
> Previews and unit/UI tests cover margins, media scaling, WebView clamping, and distribution edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 00708c8151db642fbf159fd37cb021bce75563d7. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt [hunk 1]
```diff
@@ -65,46 +66,52 @@ internal fun HorizontalStack(
...
+    modifier: Modifier = Modifier,
+    itemContent: @Composable (index: Int, item: ComponentStyle, modifier: Modifier) -> Unit,
+) {
+    ConstrainedFillLayout(
+        config = ConstrainedFillLayout.Config.Horizontal(
+            distribution = dimension.distribution,
+            arrangement = dimension.distribution.toHorizontalArrangement(spacing),
+            alignment = dimension.alignment.toAlignment(),
+        ),
+        fillConstraints = items.map { it.size.width as? Fill },
```
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt]: An invisible child emits no layout node (StackComponentView returns early), so measurables ends up shorter than fillConstraints and every child after it gets paired with the wrong constraint. Same in VerticalStack. Would parent data be a safer way to pass these?

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt [hunk 2]
```diff
@@ -65,46 +66,52 @@ internal fun HorizontalStack(
...
-    private val distribution: FlexDistribution,
-    private val spacing: Dp,
-    private val fillSpaceSpacer: @Composable (Float) -> Unit,
-    private val width: SizeConstraint,
-) : HorizontalStackScope {
-    private var hasAnyItemsWithFillWidth = false
-    val shouldApplyFillSpacers: Boolean
-        get() = width !is Fit && !hasAnyItemsWithFillWidth
-    var rowContent: @Composable RowScope.() -> Unit = {}
+private val List<ComponentStyle>.hasConstrainedFillWidth: Boolean
```
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt]: These come from the un-margined item.size, but the child's modifier enforces size.addMargin. Won't a Fill(min) child with margins force a bigger size than the parent allocated, and a Fill(max) child lose margin space from its content?

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Size.kt [hunk 3]
```diff
@@ -40,22 +39,90 @@ internal fun Modifier.size(
...
+        space = layoutWidth,
+        layoutDirection = layoutDirection,
+    )
+    val y = (verticalAlignment ?: Alignment.CenterVertically).align(
+        size = placeable.height,
+        space = layoutHeight,
+    )
+
+    layout(layoutWidth, layoutHeight) {
+        placeable.placeRelative(x, y)
```
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Size.kt]: Should this be place() instead of placeRelative()? align() already resolves layoutDirection, so leading/trailing alignment looks like it would flip twice in RTL.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt (hunk 1 above)
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt]: An invisible child renders nothing, so measurables ends up with fewer entries than fillConstraints and every child after it gets paired with the wrong constraint. Same in VerticalStack. Should this use parent data instead of matching by index?

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt (hunk 2 above)
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt]: This reads item.size, which has no margins, but the child ends up enforcing size.addMargin. So a Fill child with min or max plus margins gets allocated one size and draws another. Should this use the margined size?

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Size.kt (hunk 3 above)
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Size.kt]: Should this be place() instead of placeRelative()? align() already handles layoutDirection, so RTL looks like it gets flipped twice.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt (hunk 1 above)
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt]: what happens if one of the children is invisible? I think it emits no node, so everything after it would line up with the wrong constraint here. maybe safer to pass these through parent data than match by index?

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt (hunk 2 above)
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt]: these are the sizes without margins, but the child enforces the margined ones via addMargin. so the parent hands out one size and the child draws another? should this add margins too?

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Size.kt (hunk 3 above)
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Size.kt]: should this be place()? align() already handles layoutDirection so I think RTL ends up flipped twice

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt (hunk 1 above)
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt]: what happens if one of the children is invisible? I think it emits no node, so everything after it would line up with the wrong constraint here. maybe safer to pass these through parent data than match by index?

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt (hunk 2 above)
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt]: these are the sizes without margins, but the child enforces the margined ones via addMargin. so the parent hands out one size and the child draws another? should this add margins too?

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Size.kt (hunk 3 above)
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Size.kt]: should this be place()? align() already handles layoutDirection so I think RTL ends up flipped twice (wrapContentWidth's own node uses place() for this reason)

```suggestion
        placeable.place(x, y)
```

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt (hunk 1 above)
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt]: Wwhat happens when a child is invisible?

We implement `visible=false` as an early return so a hidden child is never composed at all. I think that means the `Layout` gets fewer measurables than this list has entries, and everything after the hidden child pairs with the wrong constraint by index?

Maybe a fix with `parentData`? 🤔

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt (hunk 2 above)
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt]: These come from `item.size`, but the child enforces `size.addMargin`.

So a `Fill(min = 50)` child with 10dp margins gets allocated 50 here but measures at 70, eating into its siblings.

Should these include margins too?

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Size.kt (hunk 3 above)
Comment (joshdholtz) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Size.kt]: Should this be `place()`? The`align()` from above already handles layoutDirection so I think RTL ends up flipped twice?

```suggestion
        placeable.place(x, y)
```

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt (hunk 1 above)
Comment (JZDesign) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/HorizontalStack.kt]: Okay, we have a pre-existing bug with our flex stacks, hidden children hide but the spacers are still laid out. I will address that in follow up PRs for both iOS and Android

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Size.kt [hunk 4]
```diff
@@ -40,22 +39,90 @@ internal fun Modifier.size(
...
+        Constraints(
+            minWidth = widthConstraints.min,
+            maxWidth = widthConstraints.max,
+            minHeight = heightConstraints.min,
+            maxHeight = heightConstraints.max,
+        ),
+    )
+
+    val layoutWidth = placeable.width.coerceIn(incomingConstraints.minWidth, incomingConstraints.maxWidth)
+    val layoutHeight = placeable.height.coerceIn(incomingConstraints.minHeight, incomingConstraints.maxHeight)
```
[the remark this answers is not in the document]
Comment (JZDesign) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/modifier/Size.kt:63]: False positive. Previews confirm overflow works and is not clipped
